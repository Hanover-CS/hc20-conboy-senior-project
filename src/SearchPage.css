.SearchList ul {
  list-style-type: disc;
}