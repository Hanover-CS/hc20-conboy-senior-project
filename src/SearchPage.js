/* Author: Patrick Conboy
 * Date Created: 11/4/2019
 * Description: This page allows the user to search our database list of anime.
 * The FilterSearchList provides most of the actual functionality code.
 */

import React, { Component } from 'react'
import FilteredSearchList from './FilteredSearchList'
import firebase from './components/Firebase/firebase'

class SearchPage extends Component {
  constructor(props) {
    super(props)

    const db = firebase.firestore()

    const animeList = []
    db.collection("anime").get().then((snapshot) => {
      snapshot.docs.forEach(doc => {
        let anime = doc.data()
        let animeName = anime['name']
        animeList.push(animeName)
      })
    })

    this.state = {animeList}
  }

  render() {
    return (
      <div className="searchList">
        <section className="section">
          <FilteredSearchList items={this.state.animeList} />
        </section>
      </div>
    )
  }
}

export default SearchPage