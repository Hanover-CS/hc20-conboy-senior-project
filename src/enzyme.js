/* Author: Patrick Conboy
 * Date Created: 1/1/2019
 * Description: This file initializes Enzyme for testing purposes.
 */

import Enzyme, { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
export { shallow, mount, render };
export default Enzyme;