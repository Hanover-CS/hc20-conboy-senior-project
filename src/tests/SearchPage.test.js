import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow, mount, render } from '../enzyme'

import SearchPage from '../SearchPage'

describe('SearchPage Tests', () => {
  test('search page renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<SearchPage />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('search page renders correctly', () => {
    const wrapper = shallow(<SearchPage props={""}/>)

    expect(wrapper.find('.props')).toBeDefined()
    expect(wrapper.find('.animeList')).toBeDefined()
  })

  test('snapshot of search page hasn\'t changed', () => {
    const tree = renderer.create(<SearchPage />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})