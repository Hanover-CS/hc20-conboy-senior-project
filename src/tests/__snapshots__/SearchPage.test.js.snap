// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage Tests snapshot of search page hasn't changed 1`] = `
<div
  className="searchList"
>
  <section
    className="section"
  >
    <div>
      <input
        className="input"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
      />
      <ul />
    </div>
  </section>
</div>
`;
